components/admin_console/schema_admin_settings.jsx

```diff
@@ -621,17 +623,24 @@ export default class SchemaAdminSettings extends React.Component {
         this.handleChange(id, s.replace('+', '-').replace('/', '_'));
     }
 
-    handleChange = (id, value) => {
-        let saveNeeded = 'config';
-        if (this.state.saveNeeded === 'permissions') {
-            saveNeeded = 'both';
+    handleChange = (id, value, submit = false) => {
+        let saveNeeded = this.state.saveNeeded;
+
+        // Only set saveNeeded & block navigation if the config is not going to immediately be saved
+        if (!submit) {
+            if (saveNeeded === 'permissions') {
+                saveNeeded = 'both';
+            } else {
+                saveNeeded = 'config';
+            }
+            this.props.setNavigationBlocked(true);
         }
-        this.setState({
-            saveNeeded,
-            [id]: value,
-        });
 
-        this.props.setNavigationBlocked(true);
+        this.setState({saveNeeded, [id]: value}, () => {
+            if (submit) {
+                this.submitSetting(null, id, value);
+            }
+        });
```

Comment:
I initially did that but decided to split it out into a different function.
Using `handleSubmit` would cause any other unsaved settings to also be saved when you set guest access to false. since the handle submit button grabs all unsaved changes from the state. I wanted the confirmation button on the guest access modal to only save that specific attribute.

Reply:
If I am not mistaken, I don't think this should be a concern. If you change any setting on the other pages and try to change page, you should get a confirmation modal stating if you like to discard your changes or not. Hence there shouldn't be any other change your saving. We could double check on this but you can confirm for yourself too (unless there is a page that doesn't do this and that I've missed).

Reply:
Maybe I'm overcomplicating this but there are other settings in the same category that can be changed and are affected by the same save button. I also feel like this might be behavior that we could want in the future for other settings and this problem may pop up again.

Reply:
> Maybe I'm overcomplicating this but there are other settings in the same category that can be changed and are affected by the same save button. I also feel like this might be behavior that we could want in the future for other settings and this problem may pop up again.

Ok that makes sense. I think I would want to wait then on @mkraft review to get his opinion as well.